src/celeritas/optical/interactor/RayleighInteractor.hh

```diff
@@ -80,25 +80,21 @@ CELER_FUNCTION Interaction RayleighInteractor::operator()(Engine& rng) const
     do
     {
         new_dir = sample_direction(rng);
-
-        auto projected_pol = dot_product(new_dir, inc_pol_);
-
-        if (soft_zero(projected_pol))
+        if (CELER_UNLIKELY(dot_product(inc_pol_, new_dir) == 0))
         {
-            // If new direction is parallel to incident polarization, then
-            // randomly sample azimuthal direction
-            new_pol = ExitingDirectionSampler{0, new_dir}(rng);
+            // In the edge case that the incident polarization and new
+            // direction are coincident, reject the sample
+            continue;
         }
-        else
+        // Project polarization onto plane perpendicular to new direction
+        new_pol = make_unit_vector(make_orthogonal(inc_pol_, new_dir));
```

Comment:
Also back to seeing this assertion failure (in the optical collector test with a single precision build):
```console
celeritas/optical/interactor/RayleighInteractor.hh:105:
celeritas: postcondition failed: soft_zero(dot_product(new_pol, new_dir))
```

Reply:
OK so perhaps even though it was orthogonalizing, it couldn't be both a unit normal and orthogonal within precision. That makes sense. I've updated the rejection so that *this* is the condition we require, rather than rejecting based on the input values.